meta-mbl-distro/classes/mbl-partitions.bbclass

```diff
@@ -24,138 +24,405 @@ inherit mbl-artifact-names
-MBL_LOG_PARTITION_SIZE_MiB ?= "128"
-MBL_LOG_PARTITION_ALIGN_KiB ?= "${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}"
+MBL_LOG_LABEL ?= "log"
+MBL_LOG_DEFAULT_SIZE_MiB ?= "128"
+MBL_LOG_MOUNT_OPTS ?= "rw,auto,noexec,nodev,nosuid,nouser,async"
+MBL_PARTITION_NAMES += "LOG"
 
 # ------------------------------------------------------------------------------
-# Scratch partition
+# Scratch partition (file system; single bank)
 # ------------------------------------------------------------------------------
 MBL_SCRATCH_MOUNT_POINT = "${MBL_SCRATCH_DIR}"
-MBL_SCRATCH_PARTITION_LABEL ?= "scratch"
-MBL_SCRATCH_PARTITION_SIZE_MiB ?= "500"
-MBL_SCRATCH_PARTITION_ALIGN_KiB ?= "${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}"
+MBL_SCRATCH_LABEL ?= "scratch"
+MBL_SCRATCH_DEFAULT_SIZE_MiB ?= "500"
+MBL_PARTITION_NAMES += "SCRATCH"
 
 # ------------------------------------------------------------------------------
-# Home partition
+# Home partition (file system; single bank)
 # ------------------------------------------------------------------------------
 MBL_HOME_MOUNT_POINT = "${MBL_HOME_DIR}"
-MBL_HOME_PARTITION_LABEL ?= "home"
-MBL_HOME_PARTITION_SIZE_MiB ?= "450"
-MBL_HOME_PARTITION_ALIGN_KiB ?= "${MBL_FLASH_ERASE_BLOCK_SIZE_KiB}"
-
+MBL_HOME_LABEL ?= "home"
+MBL_HOME_DEFAULT_SIZE_MiB ?= "450"
+MBL_PARTITION_NAMES += "HOME"
 
 # ------------------------------------------------------------------------------
-# Raw non-file system "partitions" not recorded in the partition table
+# Validate partition variable values and compute missing values
 # ------------------------------------------------------------------------------
-# For each of the BL2 and BL3 bootloaders we usually write an image file to raw
-# flash storage. Some platforms are different though:
-# * On Raspberry Pi 3, BL2 must be in the first FAT file system partition.
-# * On Platforms that have only been partially ported to use MBL's secure boot
-#   scheme, we may have other requirements, e.g. a U-Boot SPL image.
-#
-# To allow for this, we have variables to specify two bootloader images in the
-# .wks file:
-# * MBL_WKS_BOOTLOADER1_FILENAME
-# * MBL_WKS_BOOTLOADER2_FILENAME
-#
-# The names "WKS_BOOTLOADER1" and "WKS_BOOTLOADER2" have been chosen to
-# distinguish them from the TF-A concepts of "BL2" and "BL3" - in the normal
-# (and default) case, "WKS_BOOTLOADER1" will correspond with an image
-# containing TF-A BL2 and "WKS_BOOTLOADER2" will correspond with an image
-# containing TF-A BL3, OP-TEE and U-Boot, but this need not be the case.
-#
-# Each of these partitions can be removed from the .wks file by setting the
-# MBL_WKS_SKIP_BOOTLOADERx variable to "1" which causes an
-# MBL_WKS_COMMENT_IF_SKIP_BOOTLOADERx to be set to be the WKS comment leader
-# "#" so that the line for that partition gets commented out of the .wks file.
-#
-# Each of these partitions also has an "MBL_WKS_BOOTLOADERx_ALIGN_KiB" variable
-# to specify where it should live in flash.
+python __anonymous() {
+    class MaybeBankedPartition:
```

Comment:
Why not just name this class `Partition`? `MaybeBankedPartition` is a bit clunky.

Reply:
Mmm, Yeah, I don't like the name. It's called that because an object might actually represent a pair of partitions rather than a single one, so I didn't just want to call it `Partition`.

I haven't been consistent about this distinction though... `MBL_PARTITION_NAMES` only contains a single name for each pair of banked partitions and `MBL_PARTITIONS` only contains a single entry for each pair of banked partitions. I think it would complicate the implementation too much to have separate entries for each bank though.

Maybe I should just call it `Partition` and do a better job of documentation... If you have any ideas I'll be glad to hear them!